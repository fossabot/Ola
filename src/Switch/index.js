import React from 'react'
import cx from 'classnames'
import {default as PT} from 'prop-types'

const Switch = ({ children, extraClass, ...props }) => {
  return (
    <label className={cx('ola_switch', extraClass)}>
      <input type="checkbox" className="ola_switch-input" {...props} />
      <div className="ola_switch-label">
        { children && <div className="ola_switch-label-content">{ children }</div> }
      </div>
    </label>
  )
}

Switch.defaultProps = {
  label: null
}

Switch.propTypes = {
  /** Extra className */
  extraClass: PT.string,
  /** Childen nodes */
  children: PT.oneOfType([
    PT.string,
    PT.arrayOf(PT.node),
    PT.node
  ])
}

export default Switch
