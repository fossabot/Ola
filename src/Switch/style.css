.ola_switch {
    padding: var(--size-3) 0;
    display: inline-block;
}
.ola_switch-input {
    position: absolute;
    opacity: 0;
}
.ola_switch-label {
    display: flex;
    font: var(--font-callout);
    align-items: center;
}
.ola_switch-label::before {
    --color: var(--gray-xlight);

    content: "";
    display: block;
    width: 44px;
    height: 24px;
    border-radius: 12px;
    flex: 0 0 auto;
    margin-right: 10px;
    background-color: var(--color);
    background-image: url('data:image/svg+xml;charset=UTF-8,<svg width="24px" height="24px" viewBox="0 0 24 24" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns="http://www.w3.org/2000/svg"><defs><circle id="shape" cx="12" cy="12" r="10"></circle><filter x="-25.0%" y="-20.0%" width="150.0%" height="150.0%" filterUnits="objectBoundingBox" id="shadow"><feOffset dx="0" dy="1" in="SourceAlpha" result="shadowOffsetOuter1"></feOffset><feGaussianBlur stdDeviation="1.5" in="shadowOffsetOuter1" result="shadowBlurOuter1"></feGaussianBlur><feColorMatrix values="0 0 0 0 0   0 0 0 0 0   0 0 0 0 0  0 0 0 0.2 0" type="matrix" in="shadowBlurOuter1"></feColorMatrix></filter></defs><use fill="black" filter="url(%23shadow)" xlink:href="%23shape"></use><use fill="white" xlink:href="%23shape"></use></svg>');
    background-repeat: no-repeat;
    background-position: left center;
    transition-property: background-color, background-position;
    transition-duration: 200ms;
}

.ola_switch-input:checked ~ .ola_switch-label::before {
    --color: var(--brand);
    background-position: right center;
}
.ola_switch-input:focus ~ .ola_switch-label::before {
    box-shadow: 0 0 0 4px var(--brand-focus);
}
.ola_switch-input:hover ~ .ola_switch-label::before {
    --color: var(--gray-light);
}
.ola_switch-input:checked:hover ~ .ola_switch-label::before {
    --color: var(--brand-dark);
}
