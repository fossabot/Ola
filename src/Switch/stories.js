import React from 'react'
import { storiesOf } from '@storybook/react'

import Switch from './'

storiesOf('Switch')
  .add('Default', () => (
    <figure>
      <Switch>Switch</Switch>
    </figure>
  ))
  .add('With link', () => (
    <figure>
      <Switch>
        Switch <a href='#'>with link</a>
      </Switch>
    </figure>
  ))
