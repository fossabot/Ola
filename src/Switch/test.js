import React from 'react'
import Switch from './'
import renderer from 'react-test-renderer'


it('Default Switch', () => {
  const tree = renderer
    .create(
      <Switch>Label text</Switch>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Extra class Switch', () => {
  const tree = renderer
    .create(
      <Switch className="foo">Label text</Switch>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})