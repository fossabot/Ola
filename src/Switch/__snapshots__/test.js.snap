// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Switch 1`] = `
<label
  className="ola_switch"
>
  <input
    className="ola_switch-input"
    label={null}
    type="checkbox"
  />
  <div
    className="ola_switch-label"
  >
    <div
      className="ola_switch-label-content"
    >
      Label text
    </div>
  </div>
</label>
`;

exports[`Extra class Switch 1`] = `
<label
  className="ola_switch foo"
>
  <input
    className="ola_switch-input"
    label={null}
    type="checkbox"
  />
  <div
    className="ola_switch-label"
  >
    <div
      className="ola_switch-label-content"
    >
      Label text
    </div>
  </div>
</label>
`;
